model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -56,27 +56,22 @@ def nodes_to_remove(self, graph: Graph, match: dict):
         return [match['cast'].id, match['sparse_to_dense']]
 
     def replace_sub_graph(self, graph: Graph, match: dict):
+        # TODO: it requires further refactoring and improvement to provide reshape-ability
         decoder_node = match['decoder']
+        decoder_node_name = decoder_node.soft_get('name', decoder_node.id)
         graph.remove_edge(decoder_node.id, match['sparse_to_dense'].id)
         graph.remove_edge(decoder_node.id, match['cast'].id)
         match['sparse_to_dense'].replace_node(decoder_node)
 
-        # update the TensorFlow infer function for the CTCGreedyDecoder to make necessary changes with the second input
-        decoder_node['old_infer'] = decoder_node.infer
-        decoder_node.infer = __class__.tf_greedy_decoder_infer
-        return {}
-
-    @staticmethod
-    def tf_greedy_decoder_infer(node: Node):
-        sequence_length_node = node.in_node(1)
+        sequence_length_node = decoder_node.in_node(1)
         if sequence_length_node.value is None:
             raise Error('The second input to the CTCGreedyDecoder node "{}" is not constant. This case is not '
-                        'supported with the Inference Engine.'.format(node.soft_get('name')))
-        # the batch size is the dimension with index 1 for the layer CTCGreedyDecoder
-        new_value = np.ones([node.in_node(0).shape[1], sequence_length_node.value[0]])
-        new_value[:, 0] = 0
```

Comment:
How we reflect this line in the new code?

Reply:
Fixed by adding `mask_value[:, 0] = 0`. Actually, this transformation, as I said, should be refactored further because I did not get this pad it contradicts with the specification for `CTCGreedyDecoder`. But now it works somehow for some model so I leave it as is.